PilotDataPlatform/cli
# PILOT-2587: Condense deprecating github actions into a single, active…

…, and maintained one

## Summary

Replaced the unmaintained actions namely the `create-release` and `upload-release-asset` actions into the `action-gh-release` action

## JIRA Issues

PILOT-2587

## Type of Change

Please delete options that are not relevant.


## Testing

Are there any new or updated tests to validate the changes?


## Test Directions

1. Commit a unique version increment to the project definition with these changes included
2. Run the build and publish workflow. It should net you a new release identical to how we've been pushing out releases in all our microservices
